genref/main.go

```diff
@@ -0,0 +1,356 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	texttemplate "text/template"
+
+	"github.com/pkg/errors"
+	"k8s.io/gengo/parser"
+	"k8s.io/gengo/types"
+
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	flConfig  = flag.String("config", "config.yaml", "path to config file")
+	flFormat  = flag.String("format", "html", "format for output")
+	flInclude = flag.String("include", "", "API definitions to include, comma-separated list")
+	flExclude = flag.String("exclude", "", "API definitions to exclude, comma-separated list")
+	flPath    = flag.String("output", ".", "path for the output files")
+	flVerbose = flag.Bool("verbose", false, "turn on verbose output")
+)
+
+const (
+	docCommentForceIncludes = "// +gencrdrefdocs:force"
+)
+
+type generatorConfig struct {
+	// HiddenMemberFields hides fields with specified names on all types.
+	HiddenMemberFields []string `json:"hiddenMemberFields"`
+
+	// HideTypePatterns hides types matching the specified patterns from the
+	// output.
+	HideTypePatterns []string `json:"hideTypePatterns"`
```

Comment:
Curious about the reason or need to hide types?

Reply:
There are some redundant things not very useful to be included in the docs. This gives the user an option to skip them. Not heavily used in any of the APIs currently listed.